staging/src/k8s.io/apimachinery/pkg/runtime/swagger_doc_generator.go

```diff
@@ -60,24 +61,40 @@ func fmtRawDoc(rawDoc string) string {
 		}
 	}
 
-	// Ignore all lines after ---
-	rawDoc = strings.Split(rawDoc, "---")[0]
+	var isPrevLinePartOfListOrTable bool
```

Comment:
Can you explain a bit the process about how `isPrevLinePartOfListOrTable` is working/what it does? I am having trouble convincing myself this logic is the right thing to do.

Reply:
The purpose of this variable is to save the last iteration's status, which is, whether the previous line was a part of markdown (unordered/ordered) list code or table code.

I added this variable and logic to insert an [empty line](https://github.com/kubernetes/website/pull/37415/files#diff-a43cbebd3277f08387cfc17b79f3ed15ec20e73a738470a5d28dabe7747e84b3R227) between
- [the last markdown code line of unordered list](https://github.com/kubernetes/website/pull/37415/files#diff-a43cbebd3277f08387cfc17b79f3ed15ec20e73a738470a5d28dabe7747e84b3R226) and
- [the following but separate plain text paragraph](https://github.com/kubernetes/website/pull/37415/files#diff-a43cbebd3277f08387cfc17b79f3ed15ec20e73a738470a5d28dabe7747e84b3R228).

Below is the example code block and the expected procedure.

```
01	// Allowed values are "Exact" or "Equivalent".
02	//
03	// - Exact: match a request only if it exactly matches a specified rule.
04	//   For example, if deployments can be modified via apps/v1, apps/v1beta1, and extensions/v1beta1,
05	//   but "rules" only included `apiGroups:["apps"], apiVersions:["v1"], resources: ["deployments"]`,
06	//   a request to apps/v1beta1 or extensions/v1beta1 would not be sent to the webhook.
07	// - Equivalent: match a request if modifies a resource listed in rules, even via another API group or version.
08	//   For example, if deployments can be modified via apps/v1, apps/v1beta1, and extensions/v1beta1,
09	//   and "rules" only included `apiGroups:["apps"], apiVersions:["v1"], resources: ["deployments"]`,
10	//   a request to apps/v1beta1 or extensions/v1beta1 would be converted to apps/v1 and sent to the webhook.
11	// Defaults to "Equivalent"
12	// +optional
```

- Before going through 01, `isPrevLinePartOfListOrTable` == false (from L65)
- After going through 01 (which matches to L95 `else`), 
  `isPrevLinePartOfListOrTable` == false
- After going through 02 (which matches to L76 `case len(line) == 0`), 
  `isPrevLinePartOfListOrTable` == false
- After going through 03 (which matches to L83 `HasPrefix(line, "-")`), 
  `isPrevLinePartOfListOrTable` == **true** (from L86)
- After going through 04-06 (which matches to L87 `HasPrefix(line, " ")`), 
  `isPrevLinePartOfListOrTable` == **true** (from L90)
- After going through 07 (which matches to L83 `HasPrefix(line, "-")`), 
  `isPrevLinePartOfListOrTable` == **true** (from L86)
- After going through 08-10 (which matches to L87 `HasPrefix(line, " ")`), 
  `isPrevLinePartOfListOrTable` == **true** (from L90)
- After going through 11 (which matches to L92 `isPrevLinePartOfListOrTable == true`), 
  `isPrevLinePartOfListOrTable` == false (from L94)
- After going through 12 (which matches to L80 `HasPrefix(leading, "+")`),
  `isPrevLinePartOfListOrTable` == false

---

I admit that with this suggestion the logic gets somehow messy.
It would be great for us if we can get some neater implementation of this logic.